### PR TITLE
feat: parallel rendering, make ffmpeg exporter default

### PR DESCRIPTION
**Pull request by justusmattern27**

### CODE REVIEW

File: packages/renderer/server/renderVideo.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  settings: RenderVideoSettings = {},
+) => {
+  console.log('Rendering...');
+
+  const resolvedConfigPath = path.resolve(process.cwd(), configFile);
+  const projectName = settings.name ?? 'project';
+  const hiddenFolderId = uuidv4();
+
+  /**
+   * TODO: Change this in the future after more testing.
```
Comment (hkonsti) [inline packages/renderer/server/renderVideo.ts:51]: I think we can delete this todo haha

File: packages/renderer/server/renderVideo.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+      port,
+      `${projectName}-${i}`,
+      i,
+      numOfWorkers,
+      settings.range,
+      hiddenFolderId,
+    );
+
+    renderPromises.push(renderVideoOnPage(browser, server, url));
+  }
```
Comment (hkonsti) [inline packages/renderer/server/renderVideo.ts:82]: I think we're losing time here. Couldn't a browser start rendering right after it's ready? Right now the first one needs to wait for all the others before it can start.

File: packages/renderer/server/renderVideo.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+  cleanupFiles.push(
+    path.join(process.cwd(), `output/${projectName}-audio.wav`),
+  );
+  cleanupFiles.push(
+    path.join(process.cwd(), `output/${projectName}-visuals.mp4`),
+  );
+
+  for (const folder of cleanupFolders) {
+    try {
+      await fs.promises.rm(folder, {recursive: true, force: true});
```
Comment (hkonsti) [inline packages/renderer/server/renderVideo.ts:229]: We could parallelize this with an array and Promise.all(). Feel free to ignore this though, I don't think we would notice any difference.

File: packages/template/src/project.meta [hunk 4]
```diff
@@ -13,15 +13,15 @@
     "audioOffset": 0
   },
   "preview": {
-    "fps": 60,
+    "fps": 30,
```
Comment (hkonsti) [inline packages/template/src/project.meta:16]: Is this change intended?

File: packages/ui/package.json [hunk 5]
```diff
@@ -31,7 +31,7 @@
     "preact": "^10.19.2"
   },
   "devDependencies": {
-    "@preact/preset-vite": "^2.5.0",
+    "@preact/preset-vite": "2.5.0",
```
Comment (hkonsti) [inline packages/ui/package.json:34]: any reason why the `^` was removed?

**Review by hkonsti — COMMENTED** [review]
This is great work!!!

File: package.json [hunk 6]
```diff
@@ -51,14 +51,14 @@
     "prettier": "^3.1.0",
     "prettier-plugin-organize-imports": "^3.2.4",
     "typescript": "^5.2.2",
-    "vite": "^4.5.2"
```
[the remark this answers is not in the document]
Comment (justusmattern27) [reply package.json:54]: not pinning the version caused build errors for me when I rebuilt from scratch: https://github.com/preactjs/preset-vite/issues/21

File: packages/template/src/project.meta (hunk 4 above)
Comment (justusmattern27) [reply packages/template/src/project.meta:16]: yeah, this makes the preview in the editor smoother
